src/envoy/transcoding/config.cc

```diff
@@ -0,0 +1,167 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "src/envoy/transcoding/config.h"
+
+#include "contrib/endpoints/src/grpc/transcoding/json_request_translator.h"
+#include "contrib/endpoints/src/grpc/transcoding/response_to_json_translator.h"
+#include "envoy/common/exception.h"
+#include "envoy/http/filter.h"
+#include "google/protobuf/descriptor.h"
+#include "google/protobuf/descriptor.pb.h"
+#include "google/protobuf/util/type_resolver.h"
+#include "google/protobuf/util/type_resolver_util.h"
+#include "server/config/network/http_connection_manager.h"
+
+using google::api_manager::transcoding::JsonRequestTranslator;
+using google::api_manager::transcoding::RequestInfo;
+using google::api_manager::transcoding::ResponseToJsonTranslator;
+using google::api_manager::transcoding::Transcoder;
+using google::api_manager::transcoding::TranscoderInputStream;
+using google::protobuf::DescriptorPool;
+using google::protobuf::FileDescriptor;
+using google::protobuf::FileDescriptorSet;
+using google::protobuf::io::ZeroCopyInputStream;
+using google::protobuf::util::error::Code;
+using google::protobuf::util::Status;
+
+namespace Grpc {
+namespace Transcoding {
+
+namespace {
+
+const std::string kTypeUrlPrefix{"type.googleapis.com"};
+
+// Transcoder implementation based on JsonRequestTranslator &
+// ResponseToJsonTranslator
+class TranscoderImpl : public Transcoder {
+ public:
+  // request_translator - a JsonRequestTranslator that does the request
+  //                      translation
+  // response_translator - a ResponseToJsonTranslator that does the response
+  //                       translation
+  TranscoderImpl(std::unique_ptr<JsonRequestTranslator> request_translator,
+                 std::unique_ptr<ResponseToJsonTranslator> response_translator)
+      : request_translator_(std::move(request_translator)),
+        response_translator_(std::move(response_translator)),
+        request_stream_(request_translator_->Output().CreateInputStream()),
+        response_stream_(response_translator_->CreateInputStream()) {}
+
+  // Transcoder implementation
+  TranscoderInputStream* RequestOutput() { return request_stream_.get(); }
+  Status RequestStatus() { return request_translator_->Output().Status(); }
+
+  ZeroCopyInputStream* ResponseOutput() { return response_stream_.get(); }
+  Status ResponseStatus() { return response_translator_->Status(); }
+
+ private:
+  std::unique_ptr<JsonRequestTranslator> request_translator_;
+  std::unique_ptr<ResponseToJsonTranslator> response_translator_;
+  std::unique_ptr<TranscoderInputStream> request_stream_;
+  std::unique_ptr<TranscoderInputStream> response_stream_;
+};
+}
+
+Config::Config(const Json::Object& config, Server::Instance& server) {
+  std::string proto_descriptor_file = config.getString("proto_descriptor");
```

Comment:
That is expected, if you specified transcoding filter but you don't have proto_descriptor, it should crash.